backend.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// backend is client core for bitlum exchange service.
+type backend interface {
```

Comment:
Explain why we need this interface, that is hard to understand for external viewer